### PR TITLE
Common galaxy options

### PR DESCRIPTION
**Pull request by o01eg**
Extracted from #2417.

Moves galaxy options `setup.*` into common code.
Setup galaxy settings and fill AIs in multiplayer lobby according to options.

### CODE REVIEW

File: util/MultiplayerCommon.cpp [hunk 1]
```diff
@@ -38,6 +38,19 @@ namespace {
...
+        db.Add("setup.specials.frequency",                  UserStringNop("OPTIONS_DB_GAMESETUP_SPECIALS_FREQUENCY"), GALAXY_SETUP_MEDIUM, RangedValidator<GalaxySetupOption>(GALAXY_SETUP_NONE, GALAXY_SETUP_RANDOM));
+        db.Add("setup.monster.frequency",                   UserStringNop("OPTIONS_DB_GAMESETUP_MONSTER_FREQUENCY"), GALAXY_SETUP_MEDIUM, RangedValidator<GalaxySetupOption>(GALAXY_SETUP_NONE, GALAXY_SETUP_RANDOM));
+        db.Add("setup.native.frequency",                    UserStringNop("OPTIONS_DB_GAMESETUP_NATIVE_FREQUENCY"), GALAXY_SETUP_MEDIUM, RangedValidator<GalaxySetupOption>(GALAXY_SETUP_NONE, GALAXY_SETUP_RANDOM));
+        db.Add("setup.ai.player.count",                     UserStringNop("OPTIONS_DB_GAMESETUP_NUM_AI_PLAYERS"), 6, RangedValidator<int>(0, IApp::MAX_AI_PLAYERS()));
+        db.Add("setup.ai.aggression", UserStringNop("OPTIONS_DB_GAMESETUP_AI_MAX_AGGRESSION"),        MANIACAL,                   RangedValidator<Aggression>(BEGINNER, MANIACAL));
```
Comment (geoffthemedio) [inline util/MultiplayerCommon.cpp]: fix indenting

File: server/ServerFSM.cpp [hunk 2]
```diff
@@ -803,6 +803,26 @@ MPLobby::MPLobby(my_context c) :
...
+            // use AI count from option
+            const int ai_count = GetOptionsDB().Get<int>("setup.ai.player.count");
+            while (m_ai_next_index <= ai_count && (m_ai_next_index <= max_ai || max_ai < 0)) {
+                PlayerSetupData player_setup_data;
+                player_setup_data.m_player_id =     Networking::INVALID_PLAYER_ID;
+                player_setup_data.m_player_name =   UserString("AI_PLAYER") + "_" + std::to_string(m_ai_next_index++);
+                player_setup_data.m_client_type =   Networking::CLIENT_TYPE_AI_PLAYER;
+                player_setup_data.m_empire_name =   GenerateEmpireName(player_setup_data.m_player_name, m_lobby_data->m_players);
+                player_setup_data.m_empire_color =  GetUnusedEmpireColour(m_lobby_data->m_players);
+                if (m_lobby_data->m_seed != "")
```
Comment (geoffthemedio) [inline server/ServerFSM.cpp]: `!m_lobby"_data->m_seed.empty()` probably?

Comment (o01eg) [reply server/ServerFSM.cpp]: Fixed.

File: util/MultiplayerCommon.cpp (hunk 1 above)
Comment (o01eg) [reply util/MultiplayerCommon.cpp]: Fixed.

**Pull request merged**